ext/hash/php_hash_sha.h

```diff
@@ -56,10 +56,10 @@ PHP_HASH_API void PHP_SHA256Update(PHP_SHA256_CTX *, const unsigned char *, size
 void SHA256_Transform_sse2(uint32_t state[PHP_STATIC_RESTRICT 8], const uint8_t block[PHP_STATIC_RESTRICT 64], uint32_t W[PHP_STATIC_RESTRICT 64], uint32_t S[PHP_STATIC_RESTRICT 8]);
 #endif
 
-#if (defined(__i386__) || defined(__x86_64__)) && defined(HAVE_IMMINTRIN_H)
+#if ((defined(__i386__) || defined(__x86_64__)) && defined(HAVE_IMMINTRIN_H)) || ((defined(_M_X64) || defined(_M_IX86)) && defined(ZEND_INTRIN_SSSE3_RESOLVER))
 # if defined(__SSSE3__) && defined(__SHA__)
 #  define PHP_HASH_INTRIN_SHA_NATIVE 1
-# elif defined(HAVE_FUNC_ATTRIBUTE_TARGET)
+# elif defined(HAVE_FUNC_ATTRIBUTE_TARGET) || defined(_M_X64) || defined(_M_IX86)
```

Comment:
Should we define `HAVE_FUNC_ATTRIBUTE_TARGET` on Windows or MSVC, so that we can omit ` || defined(_M_X64) || defined(_M_IX86)` here?

Reply:
If we don't have an equivalent of `__attribute__((target(...)))` on MSVC, we should not define `PHP_HASH_INTRIN_SHA_RESOLVER` because `SHA256_Transform_shani()` won't compile.

Reply:
Note that `SHA256_Transform_shani()` is declared in php_hash_sha.h without any attribute. `PHP_HASH_INTRIN_SHA_RESOLVER` is still needed to dynamically resolve; this is not to be confused with a fictional `PHP_HASH_INTRIN_SHA_FUNC_PROTO`, what would imply that function attributes are supported.